pkg/backend/httpstate/client/ai.go

```diff
@@ -22,6 +22,16 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/apitype"
 )
 
+// Maximum number of characters to send to Copilot for requests.
+// We do this in "chars" to avoid including a proper token counting library for now.
+// Tokens are 3-4 characters as a rough estimate.
+const (
+	// 4kb, we're optimizing for latency here. ~1000 tokens.
+	maxCopilotSummarizeUpdateContentLength = 4000
+	// 100kb, we're optimizing for quality here. ~25000 tokens. May cull further on the backend.
```

Comment:
> May cull further on the backend

What exactly does the backend do? Should we send the entire payload and let the backend do the trimming? And then this can be relaxed once we have cheap models with large context windows without the CLI changes?

Reply:
> What exactly does the backend do?

We'll implement a similar check for now on the backend. (~25k tokens). 

> Should we send the entire payload and let the backend do the trimming?

Yes! I like this too. Or maybe better would be to send the `{project,version}` etc so if we want we can switch over to rendering on the BE.

Reply:
We don't want it to get completely out of control. We could bump this to 250kb or so. We have a 300kb limit on requests in the BE today.

Reply:
Limited to 200kb for now, to balance latency / content for the BE to play with.